frontend2/src/views/Register.tsx

```diff
@@ -1,7 +1,172 @@
-import React from "react";
+import React, { useState } from "react";
+import * as Auth from "../utils/auth";
+import Input from "../components/elements/Input";
+import Button from "../components/elements/Button";
+import { type SubmitHandler, useForm } from "react-hook-form";
+import { useCurrentUser } from "../contexts/CurrentUserContext";
+import {
+  GenderEnum,
+  type Country,
+  type CreateUserInput,
+  COUNTRIES,
+} from "../utils/apiTypes";
+import SelectMenu from "../components/elements/SelectMenu";
+import { type Maybe } from "../utils/utilTypes";
+
+const REQUIRED_ERROR_MSG = "This field is required.";
 
 const Register: React.FC = () => {
-  return <p>register page</p>;
+  const { login } = useCurrentUser();
+  const {
+    register,
+    handleSubmit,
+    setValue,
+    setError,
+    clearErrors,
+    formState: { errors },
+  } = useForm<CreateUserInput>();
+  const [gender, setGender] = useState<Maybe<GenderEnum>>();
+  const [country, setCountry] = useState<Maybe<Country>>();
+
+  const onSubmit: SubmitHandler<CreateUserInput> = async (data) => {
+    if (gender === undefined || country === undefined) {
+      return;
+    }
+    try {
+      const newUser = await Auth.register(data);
+      login(newUser);
+      console.log("logged in successfully");
+    } catch (error) {
+      console.log("failure to register", error);
```

Comment:
Consider showing a user-facing error, even just propagating the backend message up to the frontend is fine. (errors where nothing seems to happen would be annoying to user)

Reply:
and it's not clear to me that the error does show, especially if it's in the login method? If it does feel free to ignore me, I haven't locally tested yet

Reply:
I'll do this in a separate PR, as this PR is already pretty big. In that PR I'm also planning on creating a reusable component for displaying these errors.

Reply:
oooo a reusable error comment, niceeee

Reply:
Actually for this PR I can make a super simple error message and replace it in the later PR